frappe/__init__.py

```diff
@@ -1508,17 +1508,24 @@ def get_module_list(app_name):
 	return get_file_items(get_app_path(app_name, "modules.txt"))
 
 
-def get_all_apps(with_internal_apps=True, sites_path=None):
-	"""Get list of all apps via `sites/apps.txt`."""
-	if not sites_path:
-		sites_path = local.sites_path
+def get_all_apps(with_internal_apps=True, sites_path=None, site=None) -> list[str]:
+	"""
+	Function to retrieve a list of all apps via `sites/apps.txt`.
+	If a site is specified - will only return the list apps installed on the site
```

Comment:
@akhilnarang this doesn't quite make sense to me. It should maybe prefer apps installed on this site to handle conflicting modules. It shouldn't `only` return installed apps, for that we have `get_installed_apps`? 

Maybe not make it an argument but always resolve conflict using `frappe.local.site` if it's available.

Reply:
@ankush beat me to it!

Reply:
why do we want to cache data about apps that aren't installed in site-specific cache anyway?

Reply:
Makes sense - changed it such that if a site is initialized - returns the installed apps. Otherwise returns all.